### ISSUE TITLE
Web flake: "Did not find anything for request" "unknown error loading"

### ISSUE THREAD

**Issue by dnfield** [flutter/flutter#49435]
relevant portion of log:

```
[ +357 ms] Did not find anything for request: packages/collection/collection.ddc.js.errors
[   +3 ms] Did not find anything for request: packages/source_span/source_span.ddc.js.errors
[   +2 ms] Did not find anything for request: packages/string_scanner/src/eager_span_scanner.ddc.js.errors
[   +2 ms] Did not find anything for request: packages/vector_math/hash.ddc.js.errors
00:58 +121 ~2 -1: compiling /tmp/flutter sdk/packages/flutter/test/rendering/non_render_object_root_test.dart [E]                                                                                      
  Failed to load "/tmp/flutter sdk/packages/flutter/test/rendering/non_render_object_root_test.dart": Unknown error loading http://localhost:35313/packages/collection/collection.ddc.js
[ +988 ms] Did not find anything for request: gestures/assets/FontManifest.json
[   +9 ms] Did not find anything for request: gestures/packages/ui/assets/Roboto-Regular.ttf
```

https://cirrus-ci.com/task/5426469165858816?command=main#L557

This seems somewhat similar to the flake we had around image comparisons on web.

**Comment by dnfield** [comment]
Another one: https://cirrus-ci.com/task/5690696585707520?command=main#L1644

Failed on postsubmit causing tree to go red.

**Comment by jonahwilliams** [comment]
I think we should probably handle this in the test runner. detecting that test setup failed and skipping the test seems like the most reasonable thing to do.

**Comment by dnfield** [comment]
Is it something that could just be retried?

**Comment by jonahwilliams** [comment]
hard to say, because I don't think we really know what went wrong.

**Comment by jonahwilliams** [comment]
i.e. its possible after failing like this that no other tests will pass

**Comment by dnfield** [comment]
Ahh I see - so maybe just restart the whole job in this case?

**Comment by jonahwilliams** [comment]
Yeah ideally we would have a way to signal that from within the agent. In practice, I don't think that is possible today

**Comment by jonahwilliams** [comment]
Update: Dart team has landed some updates to the test package which should help with this issue. Waiting on a release, then we can update our versions and observe if the severity is reduced

**Comment by dnfield** [comment]
Anecdotally, this is happening more frequently of late, and starting to negatively impact the autorollers.

Is there any chance this can be brought in before April?  If not, we may have to consider disabling the tests, at least for autorollers.  /cc @jason-simmons

**Comment by dnfield** [comment]
Another one https://cirrus-ci.com/task/4964001691992064?command=main#L2254

https://github.com/flutter/flutter/pull/50520

**Comment by dnfield** [comment]
https://cirrus-ci.com/task/6117437137485824?command=main#L2157

**Comment by dnfield** [comment]
https://github.com/flutter/flutter/pull/50803 - engine roll containing multiple TODAY bug fixes.

**Comment by yjbanov** [comment]
I just ran all shards 15 times and got 100% success ([details](https://github.com/flutter/flutter/pull/51642)). If you have any tips for how to reproduce this I'm all ears. I'll keep trying though.

**Comment by yjbanov** [comment]
Ran it again 8 shards 30 times each (that's 240 shard runs total). [All green](https://github.com/flutter/flutter/pull/51642/checks?check_run_id=475669227). I also checked the last 10-ish Cirrus master build failures. None of them were from web tests.

I am making the conclusion that this is fixed now. It could have been fixed by one of the Dart rolls or one of `flutter update-package --force-upgrade`s (since the error is coming from a dependency).

Closing for now.
